docs/topics/sealed-classes.md

```diff
@@ -1,99 +1,350 @@
 as long as they are compatible with normal inheritance rules in Kotlin.
 
-Subclasses of sealed classes must have a proper qualified name. They can't be local nor anonymous objects.
+Subclasses of sealed classes must have a properly qualified name. They can't be local or anonymous objects.
 
-> `enum` classes can't extend a sealed class (as well as any other class), but they can implement sealed interfaces.
+> `enum` classes can't extend a sealed class, or any other class. But they can implement sealed interfaces:
 >
+> ```kotlin
+> sealed interface Error
+> 
+> // enum class extending the sealed interface Error
+> enum class ErrorType : Error {
+>    FILE_ERROR, DATABASE_ERROR
+> }
+>
+> ```
+> 
 {type="note"}
 
 These restrictions don't apply to indirect subclasses. If a direct subclass of a sealed class is not marked as sealed,
 it can be extended in any way that its modifiers allow:
 
 ```kotlin
-sealed interface Error // has implementations only in same package and module
+// Sealed interface 'Error'has implementations only in the same package and module
+sealed interface Error
 
-sealed class IOError(): Error // extended only in same package and module
-open class CustomError(): Error // can be extended wherever it's visible
+// Sealed class 'IOError' extends 'Error' and is extendable only within the same package
+sealed class IOError(): Error
+
+// Open class 'CustomError' extends 'Error' and can be extended anywhere it's visible
+open class CustomError(): Error
 ```
 
 ### Inheritance in multiplatform projects
 
 There is one more inheritance restriction in [multiplatform projects](multiplatform-get-started.md): direct subclasses of sealed classes must
-reside in the same source set. It applies to sealed classes without the [expected and actual modifiers](multiplatform-expect-actual.md).
+reside in the same [source set](multiplatform-discover-project.md#source-sets). It applies to sealed classes without the [expected and actual modifiers](multiplatform-expect-actual.md).
 
 If a sealed class is declared as `expect` in a common source set and have `actual` implementations in platform source sets,
 both `expect` and `actual` versions can have subclasses in their source sets. Moreover, if you use a hierarchical structure,
 you can create subclasses in any source set between the `expect` and `actual` declarations. 
 
 [Learn more about the hierarchical structure of multiplatform projects](multiplatform-hierarchy.md). 
 
-## Sealed classes and when expression
+## Use sealed classes with when expression
 
 The key benefit of using sealed classes comes into play when you use them in a [`when`](control-flow.md#when-expression)
-expression. 
-If it's possible to verify that the statement covers all cases, you don't need to add an `else` clause to the statement:
+expression.
+The `when` expression, used with a sealed class, allows the Kotlin compiler to check exhaustively that all possible cases are covered. 
```

Comment:
🤔 - That's a tough one. I like all 3 (the original included).
In my opinion:
1. The original is a bit wordy but conveys all the information. 
2. "The when expression with a sealed class..." It's more concise, but I feel it's also a bit more passive - it might imply that the cases are covered by default, rather than being actively checked by the compiler.
3. "If you use.." I think this one is a bit less direct in its structure, but a bit more active than the original.  

I don't think it's an issue being a bit more wordy in this case to belabor a point, because it's one of the most important functionalities of sealed classes.

Reply:
Fair enough. Up to you.